orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -362,7 +362,9 @@ def _get_exported_model(self, input_schema, *inputs, **kwargs):
                     _logger.suppress_os_stream_output(log_level=self._debug_options.logging.log_level):
                 required_export_kwargs = {'input_names': self._input_info.names,
                                           'output_names': output_names,
-                                          'opset_version': ortmodule.ONNX_OPSET_VERSION,
+                                          'opset_version': ortmodule._defined_from_envvar(
+                                                'ORTMODULE_ONNX_OPSET_VERSION',
+                                                ortmodule.ONNX_OPSET_VERSION, warn=True),
```

Comment:
I am ok with this change. Do you think we should overwrite the value in `__init__` as @thiagocrepaldi was suggesting so that if users query that value, they get the updated value that is being used in `ORTModule`?

Reply:
I don't mind but this requires a new attribute stored in the class. That's very close to what I did in my previous PR. I did not do it because of that comment: https://github.com/microsoft/onnxruntime/pull/9350#issuecomment-947349307.

Reply:
Oh, what I meant was, let's rewrite the value `ortmodule.ONNX_OPSET_VERSION` to what is set in the env var inside the __init__ constructor? And then use that overwritten value in the exporter directly.

So the constructor of graph execution manager we can do something like:
```py
ortmodule.ONNX_OPSET_VERSION = ortmodule._defined_from_envvar(
                                                'ORTMODULE_ONNX_OPSET_VERSION',
                                                ortmodule.ONNX_OPSET_VERSION, warn=True)
```

and during export, we just do
```py
required_export_kwargs = {'input_names': self._input_info.names,
                                          'output_names': output_names,
                                          'opset_version': ortmodule.ONNX_OPSET_VERSION...
```

What are your thoughts on this?

Reply:
ortmodule.ONNX_OPSET_VERSION would then keep the last opset detected in ORTModule constructor. That's not something very common. We have a last option we did not think of, it is to use method to. We could create a class OnnxOpset  write something like `ORTModule(nn).to(OnnxOpset.op14)`. That way, there is no new argument in the constructor, we still need a new attribute, but it follows torch API.

Reply:
The `.to(opset_version)` is very pythonic as it promotes decoupling and yet an easy way to push changes to the model. However according to `https://pytorch.org/docs/stable/generated/torch.nn.Module.html?highlight=#torch.nn.Module.to`, `torch.nn.Module.to()` is meant to set `device`, `dtype` or `memory_format` only. Its signature `to(*args, **kwargs)` does allow a lot more than that, so maybe it is a fair way to push other settings. That is a good idea, but let's discuss it separately from this PR, as it has usability changes for pytorch users at the same time it can be used for a lot more scenarios than just opset version.

For now, making the change that Baiju suggested makes sense. `ortmodule.ONNX_OPSET_VERSION` should always store the most current value, not the "initial" value that is ignored when the env var is set. If we create a global variable with a setting, the global shouldn't be outdated. Otherwise, there will be conflicting values. `onnxruntime.ONNX_OPSET_VERSION` and `os.environ['ORTMODULE_ONNX_OPSET_VERSION']` and an user extending ORTModule would be in doubt in which one to use

Reply:
I hope I did it the way you expected. This kind of implementation is unusual.